pydantic/_internal/_model_construction.py

```diff
@@ -137,22 +137,34 @@ def wrapped_model_post_init(self: BaseModel, __context: Any) -> None:
                 parent_parameters = getattr(cls, '__pydantic_generic_metadata__', {}).get('parameters', ())
                 parameters = getattr(cls, '__parameters__', None) or parent_parameters
                 if parameters and parent_parameters and not all(x in parameters for x in parent_parameters):
-                    combined_parameters = parent_parameters + tuple(x for x in parameters if x not in parent_parameters)
-                    parameters_str = ', '.join([str(x) for x in combined_parameters])
-                    generic_type_label = f'typing.Generic[{parameters_str}]'
-                    error_message = (
-                        f'All parameters must be present on typing.Generic;'
-                        f' you should inherit from {generic_type_label}.'
-                    )
-                    if Generic not in bases:  # pragma: no cover
-                        # We raise an error here not because it is desirable, but because some cases are mishandled.
-                        # It would be nice to remove this error and still have things behave as expected, it's just
-                        # challenging because we are using a custom `__class_getitem__` to parametrize generic models,
-                        # and not returning a typing._GenericAlias from it.
-                        bases_str = ', '.join([x.__name__ for x in bases] + [generic_type_label])
-                        error_message += (
-                            f' Note: `typing.Generic` must go last: `class {cls.__name__}({bases_str}): ...`)'
+                    missing_parameters = tuple(x for x in parameters if x not in parent_parameters)
+                    if 'RootModelRootType' in tuple(
+                        x.__name__ for x in parent_parameters
+                    ) and 'RootModelRootType' not in tuple(x.__name__ for x in parameters):
```

Comment:
Couldn't you do something like?
```suggestion
                    if (
                        RootModelRootType in parent_parameters
                        and RootModelRootType not in parameters
                    ):
```
where `RootModelRootType` is `pydantic.root_model.RootModelRootType`?